src/wavm/WAVMWasmModule.cpp

```diff
@@ -565,9 +571,11 @@ Runtime::Instance* WAVMWasmModule::createModuleInstance(
           boundUser, boundFunction, sharedModulePath);
 
         // Provision the memory for the new module plus two guard regions
-        createMemoryGuardRegion();
-        Uptr newMemory = mmapPages(DYNAMIC_MODULE_MEMORY_PAGES);
-        createMemoryGuardRegion();
+        uint32_t memSize = DYNAMIC_MODULE_MEMORY_SIZE + (2 * GUARD_REGION_SIZE);
+        Uptr newMemory = growMemory(memSize);
+        createMemoryGuardRegion(newMemory);
+        createMemoryGuardRegion(newMemory + DYNAMIC_MODULE_MEMORY_SIZE +
```

Comment:
Note that before, creating a stack with memory guards either side performed three memory mappings, now we do one, and apply the read-only protection on the guard pages afterwards.